modules/telemetry/telemetry/src/log.cpp

```diff
@@ -28,36 +30,49 @@ class Logger final {
   auto operator=(const Logger&) -> Logger& = delete;
   auto operator=(Logger&&) -> Logger& = delete;
 
-  static void registerSink(std::unique_ptr<heph::telemetry::ILogSink> sink);
+  static void registerSink(std::unique_ptr<ILogSink> sink);
 
-  static void log(heph::telemetry::LogEntry&& log_entry);
+  static void log(LogEntry&& log_entry);
 
 private:
   [[nodiscard]] static auto instance() -> Logger&;
 
   /// @brief Do the actual logging. Take in LogEntry and send it to all sinks
   /// @param LogEntry, class that takes in the structured logs and formats them.
   /// @return void
-  void processLogEntries(const heph::telemetry::LogEntry& entry);
+  void processEntry(const LogEntry& entry);
+
+  /// @brief Empty the queue so that remaining messages get processed
+  void emptyQueue();
 
 private:
   absl::Mutex sink_mutex_;
   // This is used as registry for the log sinks
-  std::vector<std::unique_ptr<heph::telemetry::ILogSink>> sinks_ ABSL_GUARDED_BY(sink_mutex_);
-  heph::concurrency::MessageQueueConsumer<heph::telemetry::LogEntry> log_entries_consumer_;
+  std::vector<std::unique_ptr<ILogSink>> sinks_ ABSL_GUARDED_BY(sink_mutex_);
+  heph::containers::BlockingQueue<LogEntry> entries_;
+  std::future<void> stop_signal_;
 };
 
-Logger::Logger()
-  : log_entries_consumer_([this](const heph::telemetry::LogEntry& entry) { processLogEntries(entry); },
-                          std::nullopt) {
-  log_entries_consumer_.start();
+Logger::Logger() : entries_{ std::nullopt } {
+  stop_signal_ = std::async(std::launch::async, [this]() {
```

Comment:
I never know how to properly name future in the case where it is of type void, but maybe I would go for `message_process_future_`